text/0000-forget-marker-trait.md

```diff
@@ -0,0 +1,803 @@
+- Feature Name: `forget_marker_trait`
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+<!-- todo: Replace with RFC PR later -->
+[`local_default_bounds`]: https://github.com/rust-lang/rfcs/pull/3783
+
+# Summary
+[summary]: #summary
+
+Add a `Forget` marker trait indicating whether it is safe to skip the destructor before the value of a type exits the scope and basic utilities to work with `!Forget` types. Introduce a seamless migration route for the standard library and ecosystem.
+
+# Motivation
+[motivation]: #motivation
+
+Many readers may find the biggest problem with `Forget` to be migration. RFC's confidence is taken from the fact that migration can be done easily. See [#migration](#migration) section for details.
+
+In short, the lack of `!Forget` types undermines lifetimes, sacrificing all performance, ergonomics and efficiency. Rust, as well as external APIs, naturally converge towards `!Forget` types, but old decisions force us to shift to `Arc`, `'static`, allocations, etc.
+
+---
+
+Back in 2015, the [decision was made][safe-mem-forget] to make `mem::forget` safe, making every type implicitly implement `Forget`. All APIs in `std` could've been preserved after that change, except one. Today is 2025 and some things changed, old reasoning is no longer true. This RFC is not targeted at resource leaks in general but is instead focused on allowing a number of APIs to become safe by providing new unsafe guarantees.
```

Comment:
It's unusual to say that something that didn't exist yet was implicitly done. I'm not sure what you meant by the middle two sentences, but I've tried to make it clearer.

```suggestion
Back in 2015, the [decision was made][safe-mem-forget] to make `mem::forget` safe, making every type effectively implement `Forget`. By making `mem::forget` safe, other `std` APIs had to become unsafe, or their safe implementations became complex (such as async Rust). This RFC is not targeted at resource leaks in general, but is instead focused on allowing a number of APIs to become safe, by providing new unsafe guarantees using `Forget`.
```